client/logmon/client.go

```diff
@@ -24,12 +27,18 @@ func (c *logmonClient) Start(cfg *LogConfig) error {
 		StdoutFifo:     cfg.StdoutFifo,
 		StderrFifo:     cfg.StderrFifo,
 	}
-	_, err := c.client.Start(context.Background(), req)
+	ctx, cancel := context.WithTimeout(context.Background(), logmonRPCTimeout)
+	defer cancel()
+
+	_, err := c.client.Start(ctx, req)
 	return grpcutils.HandleGrpcErr(err, c.doneCtx)
```

Comment:
Is there any special handling we should do here if the err is from the context timeout?

Reply:
Don't think so; the logmon hook will kill plugin and re-attempt to start a new logmon: https://github.com/hashicorp/nomad/blob/v0.10.2/client/allocrunner/taskrunner/logmon_hook.go#L104-L121 .